src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -107,6 +107,43 @@ const createAppServiceInstalls = async (
 		},
 	});
 
+const deleteServiceInstallsForCurrentApp = (
+	api: sbvrUtils.PinejsClient,
+	deviceIds: number[],
+) =>
+	api.delete({
+		resource: 'service_install',
+		options: {
+			$filter: {
+				device: { $in: deviceIds },
+				installs__service: {
+					$any: {
+						$alias: 's',
+						$expr: {
+							s: {
+								application: {
+									$any: {
+										$alias: 'a',
+										$expr: {
+											a: {
+												owns__device: {
+													$any: {
+														$alias: 'd',
+														$expr: { d: { id: { $in: deviceIds } } },
```

Comment:
I modified this to avoid needing to separately query the app ids but it is otherwise extracted from https://github.com/balena-io/open-balena-api/pull/504